devtools/maven/src/main/java/io/quarkus/maven/AddExtensionMojo.java

```diff
@@ -49,41 +49,30 @@ protected void validateParameters() throws MojoExecutionException {
     @Override
     public void doExecute(BuildFile buildFile) throws MojoExecutionException {
 
-        boolean closeBuildFile = false;
-        try {
-            if (buildFile == null) {
-                try {
-                    buildFile = BuildTool.MAVEN.createBuildFile(new FileProjectWriter(project.getBasedir()));
-                } catch (IOException e) {
-                    throw new MojoExecutionException("Failed initialize project configuration tools", e);
-                }
-            }
-            Set<String> ext = new HashSet<>();
-            if (extensions != null && !extensions.isEmpty()) {
-                ext.addAll(extensions);
-            } else {
-                // Parse the "extension" just in case it contains several comma-separated values
-                // https://github.com/quarkusio/quarkus/issues/2393
-                ext.addAll(Arrays.stream(extension.split(",")).map(s -> s.trim()).collect(Collectors.toSet()));
-            }
-
+        if (buildFile == null) {
             try {
-                final AddExtensionResult result = new AddExtensions(buildFile)
-                        .addExtensions(ext.stream().map(String::trim).collect(Collectors.toSet()));
-                if (!result.succeeded()) {
-                    throw new MojoExecutionException("Unable to add extensions");
-                }
+                buildFile = BuildTool.MAVEN.createBuildFile(new FileProjectWriter(project.getBasedir()));
             } catch (IOException e) {
-                throw new MojoExecutionException("Unable to update the pom.xml file", e);
+                throw new MojoExecutionException("Failed initialize project configuration tools", e);
             }
-        } finally {
```

Comment:
@aloubyansky I need you to take a look at this one. Right now, removing this code doesn't change the plugin behaviour, but I'm pretty sure that's not the way we'd want to fix this class.

Reply:
Yes, it's not about just fixing this class, it's about fixing the tools. I think it's ok to merge it. The `buildFile` is closed in AddExtensions.